calculator.py

```diff
@@ -1,84 +1,225 @@
+proper pepage.
 
-  * Addition
-  * Subtractions
-  * Multiplication
-  * Division
-
-Your users should be able to send appropriate requests and get back
-proper responses. For example, if I open a browser to your wsgi
-application at `http://localhost:8080/multiple/3/5' then the response
-body in my browser should be `15`.
-
-Consider the following URL/Response body pairs as tests:
+Also added a Modulus operation because why not...
+"""
 
-```
-  http://localhost:8080/multiply/3/5   => 15
-  http://localhost:8080/add/23/42      => 65
-  http://localhost:8080/subtract/23/42 => -19
-  http://localhost:8080/divide/22/11   => 2
-  http://localhost:8080/               => <html>Here's how to use this page...</html>
-```
+import functools
+import operator
+from loguru import logger
 
-To submit your homework:
 
-  * Fork this repository (Session03).
-  * Edit this file to meet the homework requirements.
-  * Your script should be runnable using `$ python calculator.py`
-  * When the script is running, I should be able to view your
-    application in my browser.
-  * I should also be able to see a home page (http://localhost:8080/)
-    that explains how to perform calculations.
-  * Commit and push your changes to your fork.
-  * Submit a link to your Session03 fork repository!
+# These are all the same aside from the operator call.  Probably a way to combine all
+#   into one function with a lookup dictionary...
 
 
+def add(*args):
+    """ Returns a STRING with the sum of the arguments """
+    logger.debug("Entering add")
+    try:
+        return str(functools.reduce(operator.add, list(map(float, args))))
+    except ValueError:
+        return error_message(*args)
+
+
+def subtract(*args):
+    """ Returns a STRING with the difference of the arguments """
+    logger.debug("Entering subtract")
+    try:
+        return str(functools.reduce(operator.sub, list(map(float, args))))
+    except ValueError:
+        return error_message(*args)
+
+
+def multiply(*args):
+    """ Returns a STRING with the product of the arguments """
+    logger.debug("Entering multiply")
+    try:
+        return str(functools.reduce(operator.mul, list(map(float, args))))
+    except ValueError:
+        return error_message(*args)
+
+
+def divide(*args):
+    """ Returns a STRING with the quotient of the arguments """
+    logger.debug("Entering divide")
+    try:
+        return str(functools.reduce(operator.truediv, list(map(float, args))))
+    except ValueError:
+        return error_message(*args)
+    except ZeroDivisionError:
+        return "<h2>Zero Division Error</h2>"
+
+
+def modulus(*args):
+    """ Returns a STRING with the remainder of the arguments """
+    logger.debug("Entering modulus")
+    try:
+        return str(functools.reduce(operator.mod, list(map(float, args))))
+    except ValueError:
+        return error_message(*args)
+    except ZeroDivisionError:
+        return "<h2>Zero Division Error</h2>"
```

Comment:
So notice that you have a lot of duplication here. You can abstract out all your operator functions like so:
```
def operator_func(op_func, *args):
    logger.debug(f"Entering {op_func.__name__}")
    try:
        return str(functools.reduce(op_func, list(map(float, args))))
    except ValueError:
        return error_message(*args)
    except ZeroDivisionError:
        return "<h2>Zero Division Error</h2>"

def add(args):
    operator_func(operator.add, *args)

def subtract(args):
    operator_func(operator.sub, *args)
```
etc...

This way should you need to make a change to that function, i.e. catching a new error or logging some other output, you don't have to update every function.